dev_guide/builds.adoc

```diff
@@ -191,6 +191,77 @@ See the xref:../creating_images/s2i.adoc#creating-images-s2i[S2I Requirements] t
 on how to create a builder image supporting incremental builds.
 ====
 
+
+[[extended_builds]]
+
+=== Extended builds
+
+For compiled languages (Go, C, C++, Java, etc.) the dependencies necessary for
+compilation might dramatically increase the size of the image or introduce
+vulnerabilities that can be exploited.
+
+To avoid these problems, source-to-image introduces a dual image build process
+that allows an application to be built as usual - in the builder image, and
+then inject - in the runtime image, the resulted binary artifacts into the
+desired location to be executed.
+
+In order to offer even more flexibility in this process, S2I will execute an
+`assemble-runtime` script inside the runtime image that will allow you to
```

Comment:
@php-coder ?  assemble-runtime is what is executed in the runtime image for manipulating the injected artifacts, right?